### PR TITLE
Add AMQP Exchange Name decision record

### PR DESCRIPTION
**Pull request by vitbaq**
The ARQ-003 proposal aims to define the names of Exchanges to be used on
KNoT ecosystem.

Signed-off-by: Vitor Barros <vba@cesar.org.br>

### CODE REVIEW

File: decisions/architecture/ARC-003-amqp-exc-name.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ARC-003: AMQP Exchange name
+
+## Status
+Proposed
+
+## Context
+
+As defined by CloudAMQP Blog, "exchanges are message routing agents, defined by the virtual host within RabbitMQ. An exchange is responsible for routing the messages to different queues with the help of header attributes, bindings, and routing keys."
+The Exchange name needs to be defined taking into account the architecture defined for the system. Thus, there will be a client that communicates with BabelTower, which has the Connector as one of its services (in this case BabelTower is the Connector client or the client communicates directly with the Connector.
```
Comment (ramonhpr) [inline decisions/architecture/ARC-003-amqp-exc-name.md]:
```suggestion
The Exchange name needs to be defined taking into account the architecture defined for the system. Thus, there will be a client that communicates with BabelTower, which has the Connector as one of its services (in this case BabelTower is the Connector client or the client communicates directly with the Connector).
```

File: decisions/architecture/ARC-003-amqp-exc-name.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+The goal is to define names that make it clear what this Exchange is used for.
+
+## Decisions
+
+* The communication between two components will be done with two queues, where one component is consumer in one and producer in another and the other component follows the opposite.
+* The reference for choosing the queue name will be the client.
+* The queue where BabelTower consumes messages will be called 'fogOut'.
+* The queue where BabelTower consumes messages will be called 'fogIn'.
+* The queue where Connector consumes messages will be called 'ConnOut'.
```
Comment (ramonhpr) [inline decisions/architecture/ARC-003-amqp-exc-name.md]:
```suggestion
* The queue where Connector publishes messages will be called 'connOut'.
```

File: decisions/architecture/ARC-003-amqp-exc-name.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* The queue where Connector consumes messages will be called 'ConnIn'.
```
Comment (ramonhpr) [inline decisions/architecture/ARC-003-amqp-exc-name.md]:
```suggestion
* The queue where Connector consumes messages will be called 'connIn'.
```

File: decisions/architecture/ARC-003-amqp-exc-name.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Consequences
+
+These decisions are sufficient to understand the meaning of each queue. Following this recommendation requires a change in the systems involved to use the new nomenclature.
```
Comment (ramonhpr) [inline decisions/architecture/ARC-003-amqp-exc-name.md]: Add the consequence that the client should know to which server it is communicating with.
The connector or the fog.

File: decisions/architecture/ARC-003-amqp-exc-name.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+As defined by CloudAMQP Blog, "exchanges are message routing agents, defined by the virtual host within RabbitMQ. An exchange is responsible for routing the messages to different queues with the help of header attributes, bindings, and routing keys."
+The Exchange name needs to be defined taking into account the architecture defined for the system. Thus, there will be a client that communicates with BabelTower, which has the Connector as one of its services (in this case BabelTower is the Connector client or the client communicates directly with the Connector.
+The goal is to define names that make it clear what this Exchange is used for.
+
+## Decisions
+
+* The communication between two components will be done with two queues, where one component is consumer in one and producer in another and the other component follows the opposite.
+* The reference for choosing the queue name will be the client.
+* The queue where BabelTower consumes messages will be called 'fogOut'.
```
Comment (ramonhpr) [inline decisions/architecture/ARC-003-amqp-exc-name.md]:
```suggestion
* The queue where BabelTower publishes messages will be called 'fogOut'.
```

File: decisions/architecture/ARC-003-amqp-exc-name.md (hunk 1 above)
Comment (vitbaq) [reply decisions/architecture/ARC-003-amqp-exc-name.md]: Ok

File: decisions/architecture/ARC-003-amqp-exc-name.md (hunk 2 above)
Comment (vitbaq) [reply decisions/architecture/ARC-003-amqp-exc-name.md]: ok

File: decisions/architecture/ARC-003-amqp-exc-name.md (hunk 3 above)
Comment (vitbaq) [reply decisions/architecture/ARC-003-amqp-exc-name.md]: ok

File: decisions/architecture/ARC-003-amqp-exc-name.md (hunk 4 above)
Comment (vitbaq) [reply decisions/architecture/ARC-003-amqp-exc-name.md]: ok

File: decisions/architecture/ARC-003-amqp-exc-name.md (hunk 5 above)
Comment (vitbaq) [reply decisions/architecture/ARC-003-amqp-exc-name.md]: ok

File: decisions/architecture/ARC-003-amqp-exc-name.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+* The communication between two components will be done with two queues, where one component is consumer in one and producer in another and the other component follows the opposite.
+* The reference for choosing the queue name will be the client.
+* The queue where BabelTower publishes messages will be called 'fogOut'.
+* The queue where BabelTower consumes messages will be called 'fogIn'.
+* The queue where Connector publishes messages will be called 'connOut'.
+* The queue where Connector consumes messages will be called 'connIn'.
+
+## Consequences
+
+These decisions are sufficient to understand the meaning of each queue. Following this recommendation requires a change in the systems involved to use the new nomenclature and now the client should know to which server it is communicating with (e.g. the Connector or the fog).
```
Comment (ramonhpr) [inline decisions/architecture/ARC-003-amqp-exc-name.md]: Use `Babeltower` rather than `fog` as you referenced in Context

**Comment by joaoaneto** [comment]
Except for the change requested by @ramonhpr, LGTM. Thanks. :)

File: decisions/architecture/ARC-003-amqp-exc-name.md (hunk 6 above)
Comment (vitbaq) [reply decisions/architecture/ARC-003-amqp-exc-name.md]: Done

**Pull request merged**